File: Python/Array/Maximum Product Subarray.py
def maxProduct(self,arr, n):
        maxp=float("-inf")
        prod=1
        for i in range(n):
            prod*=arr[i]
            maxp=max(prod, maxp)
            if prod==0:
                prod=1
        prod=1            
        for i in range(n-1,-1,-1):
            prod*=arr[i]
            maxp=max(prod, maxp)
            if prod==0:
                prod=1
        return maxp 

File: C++/Array/Maximum Product Subarray.cpp
class Solution{
public:

	// Function to find maximum product subarray
	long long maxProduct(int *arr, int n) {
	    // code here
	    long long int maxp = INT_MIN, prod = 1;
	    for(int i=0;i<n;i++){
	        prod *= arr[i];
	        maxp = max(maxp, prod);
	        if(prod == 0){
	            prod = 1;
	        }
	    }
	    
	    prod = 1;
	    for(int i=n-1;i>=0;i--){
	        prod *= arr[i];
	        maxp = max(maxp, prod);
	        if(prod == 0){
	            prod = 1;
	        }
	    }
	    
	    return maxp;
	}
};
